lib/gpx/geojson.rb

```diff
@@ -0,0 +1,222 @@
+                'or \':geojson_data\' to convert_to_gpx'
+        end
+      end
+
+      def parse_geojson_data_from_file(filename)
+        parse_geojson_data(IO.read(filename))
+      end
+
+      def parse_geojson_data(data)
+        JSON.parse(data)
+      end
+
+      def add_tracks_to(gpx_file, geojson)
+        tracks = [line_strings_to_track(geojson)] +
+                 multi_line_strings_to_tracks(geojson)
+        tracks.reject!(&:nil?)
+        gpx_file.tracks += tracks
+        gpx_file.tracks.each { |t| gpx_file.update_meta_data(t) }
+      end
+
+      def add_waypoints_to(gpx_file, geojson)
+        gpx_file.waypoints +=
+          points_to_waypoints(geojson, gpx_file) +
+          multi_points_to_waypoints(geojson, gpx_file)
+      end
+
+      # Converts GeoJSON 'LineString' features.
+      # Current strategy is to convert each LineString into a
+      # Track Segment, returning a Track for all LineStrings.
+      #
+      def line_strings_to_track(geojson)
+        line_strings = line_strings_in(geojson)
+        return nil unless line_strings.any?
+        track = GPX::Track.new
+        line_strings.each do |ls|
+          coords = ls['geometry']['coordinates']
+          track.append_segment(coords_to_segment(coords))
+        end
+        track
+      end
+
+      # Converts GeoJSON 'MultiLineString' features.
+      # Current strategy is to convert each MultiLineString
+      # into a Track, with each set of LineString coordinates
+      # within a MultiLineString a Track Segment.
+      #
+      def multi_line_strings_to_tracks(geojson)
+        tracks = []
+        multi_line_strings_in(geojson).each do |mls|
+          track = GPX::Track.new
+          mls['geometry']['coordinates'].each do |coords|
+            seg = coords_to_segment(coords)
+            seg.track = track
+            track.append_segment(seg)
+          end
+          tracks << track
+        end
+        tracks
+      end
+
+      # Converts GeoJSON 'Point' features.
+      # Current strategy is to convert each Point
+      # feature into a GPX waypoint.
+      #
+      def points_to_waypoints(geojson, gpx_file)
+        waypoints = []
+        points_in(geojson).each do |pt|
+          coords = pt['geometry']['coordinates']
+          waypoints << point_to_waypoint(coords, gpx_file)
+        end
+        waypoints
+      end
+
+      # Converts GeoJSON 'MultiPoint' features.
+      # Current strategy is to convert each coordinate
+      # point in a MultiPoint to a GPX waypoint.
+      #
+      # NOTE: It is debatable that a MultiPoint feature
+      # might translate best into a GPX route, which is
+      # described as
+      #   "an ordered list of waypoints representing a
+      #    series of turn points leading to a destination."
+      # See http://www.topografix.com/gpx/1/1/#type_rteType
+      #
+      def multi_points_to_waypoints(geojson, gpx_file)
+        waypoints = []
+        multi_points_in(geojson).each do |mpt|
```

Comment:
Good to go. I'll try to work in these changes this weekend or next week. Thanks for the feedback.